mycpp/pass_state.py

```diff
@@ -568,6 +568,9 @@ def ComputeMinimalStackRoots(cfgs: dict[str, ControlFlowGraph],
     that can be queried by cppgen_pass.
     """
     DumpControlFlowGraphs(cfgs, facts_dir=facts_dir)
+    # The facts files can be pretty large. Sync them first to avoid reading
+    # truncated files from the solver.
+    subprocess.run('sync {}/*.facts'.format(facts_dir), shell=True)
```

Comment:
hm this seems weird

I don't think there is any situation where after calling `f.close()` you can't read the right thing back in a different process

that would be either  a Python bug or a kernel bug

Reply:
Oh you know what. I bet this is actually a case of parallel instances of mycpp-souffle clobbering each other's output. I was seeing this failure in one of the CI jobs that builds cxx-opt and cxx-asan? This could be addressed by giving each one a unique tempdir for the fact outputs. I'll confirm that and tidy this up later